src/main/scala/mrpowers/jodie/OperationMetricHelper.scala

```diff
@@ -0,0 +1,179 @@
+
+case class OperationMetricHelper(
+    path: String,
+    startingVersion: Long = 0,
+    endingVersion: Option[Long] = None
+) {
+  val spark                 = SparkSession.active
+  val deltaLog              = DeltaLog.forTable(spark, path)
+  private val metricColumns = Seq("version", "deleted", "inserted", "updated", "source_rows")
+
+  def getMetricsAsDF(): DataFrame = {
+    import spark.implicits._
+    getTransactionMetrics.toDF(metricColumns: _*)
+  }
+
+  def getMetricsAsDF(partitionCondition: String): DataFrame = {
+    import spark.implicits._
+    getTransactionMetrics(partitionCondition).toDF(metricColumns: _*)
+  }
+
+  def getTransactionMetrics: Seq[(Long, Long, Long, Long, Long)] = transformMetric(
+    generateMetric(deltaLog.history.getHistory(startingVersion, endingVersion))
+  )
+
+  def getTransactionMetrics(
+      partitionCondition: String
+  ): Seq[(Long, Long, Long, Long, Long)] = transformMetric(
+    generateMetric(
+      deltaLog.history
+        .getHistory(startingVersion, endingVersion)
+        .filter(x => filterHistoryByPartition(x, partitionCondition)),
+      Some(partitionCondition)
+    )
+  )
+
+  def getWriteMetricByPartition(
+      partitionCondition: String,
+      version: Long
+  ): Long = {
+    import spark.implicits._
+    val trimmed  = partitionCondition.trim
+    val splitCondition =
+      if (trimmed.contains(" and "))
+        trimmed.split(" and ").toSeq
+      else Seq(trimmed)
+    val conditions = splitCondition.map(x => {
+      val kv = x.split("=")
+      assert(kv.size == 2)
+      s"${kv.head.trim}=${kv.tail.head.trim.stripPrefix("\'").stripSuffix("\'")}"
+    })
+    val jsonSchema = new StructType()
+      .add("numRecords", LongType)
+      .add("minValues", MapType(StringType, StringType))
+      .add("maxValues", MapType(StringType, StringType))
+      .add("nullCount", MapType(StringType, StringType))
+    spark.read
+      .json(FileNames.deltaFile(deltaLog.logPath, version).toString)
+      .withColumn("stats", org.apache.spark.sql.functions.from_json(col("add.stats"), jsonSchema))
+      .select("add.path", "stats")
+      .map(x => {
+        val path = x.getAs[String]("path")
+        conditions.map(x => path != null && path.contains(x)).reduceOption(_ && _) match {
+          case None => 0L
+          case Some(bool) =>
+            if (bool)
+              x.getAs[String]("stats").asInstanceOf[GenericRowWithSchema].getAs[Long]("numRecords")
+            else 0L
+        }
+      })(Encoders.scalaLong)
+      .reduce(_ + _)
+  }
+
+  def transformMetric(
+      metric: Seq[(Long, OperationMetrics)]
+  ): Seq[(Long, Long, Long, Long, Long)] = metric
+    .filter(x =>
+      x._2.isInstanceOf[MergeMetric] || x._2.isInstanceOf[WriteMetric] || x._2
+        .isInstanceOf[DeleteMetric]
+    )
+    .map(x => {
+      val (deleted, inserted, updated, sourceRows) = x._2 match {
+        case MergeMetric(a, b, c, d, e, f, g, h, i, j, k, l) => (b, e, g, j)
```

Comment:
Here be explicit in naming when extracting data from Metric objects, and use wildcards for non-used attributes. This is for code readability. ie:` MergeMetric(_, deletedRows, _, _, insertedRows, _, updatedRows, _, _, sourceRows, _, _) => (deletedRows, insertedRows, updatedRows, sourceRows)`

Reply:
done and refactored